src/Terrain/GameMap.java

```diff
@@ -23,65 +23,75 @@ public GameMap()
 				// Make a ring of water around the edges, and the rest plains to start.
 				if(w == 0 || h == 0 || (w == mapWidth - 1) || (h == mapHeight - 1) )
 				{
-					map[w][h] = Tile.getTile(Tile.Terrains.WATER, Tile.Weathers.CLEAR);
+					map[w][h] = new Location(Environment.getTile(Environment.Terrains.WATER, Environment.Weathers.CLEAR));
 				}
 				else
 				{
-					map[w][h] = Tile.getTile(Tile.Terrains.PLAIN, Tile.Weathers.CLEAR);
+					map[w][h] = new Location(Environment.getTile(Environment.Terrains.PLAIN, Environment.Weathers.CLEAR));
 				}
 			}
 		}
-
+		
 		// Throw down an HQ and a factory for each.
-		map[1][8] = Tile.getTile(Tile.Terrains.HQ, Tile.Weathers.CLEAR);
-		map[1][7] = Tile.getTile(Tile.Terrains.FACTORY, Tile.Weathers.CLEAR);
-		map[2][7] = Tile.getTile(Tile.Terrains.FACTORY, Tile.Weathers.CLEAR);
+		map[1][8] = new Location(Environment.getTile(Environment.Terrains.HQ, Environment.Weathers.CLEAR));
```

Comment:
For all the locations that we are replacing, we don't want to re-create the location - just call setEnvironment().